homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,479 @@
+                hass.add_job(
+                    async_add_entities,
+                    [
+                        FritzBoxDeflectionSwitch(
+                            fritzbox_tools, entry.title, dict_of_deflection
+                        )
+                    ],
+                )
+
+    def _create_port_switches():
+
+        _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_PORTFORWARD)
+
+        service_name = "Layer3Forwarding"
+        connection_type = service_call_action(
+            fritzbox_tools, service_name, True, "GetDefaultConnectionService"
+        )
+        if not connection_type:
+            _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_PORTFORWARD)
+            return
+
+        # Return NewDefaultConnectionService sample: "1.WANPPPConnection.1"
+        connection_type = connection_type["NewDefaultConnectionService"][2:][:-2]
+
+        # Query port forwardings and setup a switch for each forward for the current device
+        port_forwards_count: int = service_call_action(
+            fritzbox_tools, connection_type, True, "GetPortMappingNumberOfEntries"
+        )["NewPortMappingNumberOfEntries"]
+
+        _LOGGER.debug(
+            "Specific %s response: GetPortMappingNumberOfEntries=%s",
+            SWITCH_TYPE_PORTFORWARD,
+            port_forwards_count,
+        )
+
+        for i in range(port_forwards_count):
+
+            portmap = service_call_action(
+                fritzbox_tools,
+                connection_type,
+                True,
+                "GetGenericPortMappingEntry",
+                NewPortMappingIndex=i,
+            )
+
+            if not portmap:
+                return None
+
+            _LOGGER.debug(
+                "Specific %s response: GetGenericPortMappingEntry=%s",
+                SWITCH_TYPE_PORTFORWARD,
+                portmap,
+            )
+
+            # We can only handle port forwards of the given device
+            local_ip = async_get_source_ip(fritzbox_tools.host)
+            _LOGGER.debug("IP source for %s is %s", fritzbox_tools.host, local_ip)
+            if portmap["NewInternalClient"] == local_ip:
+                hass.add_job(
+                    async_add_entities,
+                    [
+                        FritzBoxPortSwitch(
+                            fritzbox_tools,
+                            entry.title,
+                            portmap,
+                            i,
+                            connection_type,
+                        )
+                    ],
+                )
+
+    def _create_profile_switches():
+        _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEVICEPROFILE)
+        if len(fritzbox_tools.fritz_profiles) > 0:
+            for profile in fritzbox_tools.fritz_profiles.keys():
+                hass.add_job(
+                    async_add_entities,
+                    [FritzBoxProfileSwitch(fritzbox_tools, entry.title, profile)],
+                )
```

Comment:
I think fritz_profiles is already equal to the list you are creating in this for loop, isn't it?